cli/cli.go

```diff
@@ -29,86 +33,180 @@ func main() {
+	proxies, err := t.Proxies()
+	if err != nil {
+		fmt.Println("Failed to retrieve proxies: ", err)
+		os.Exit(1)
 	}
 
-	app.Run(os.Args)
+	var proxyNames []string
+	for proxyName := range proxies {
+		proxyNames = append(proxyNames, proxyName)
+	}
+	sort.Strings(proxyNames)
+
+	fmt.Fprintf(os.Stderr, "%sListen\t\t%sUpstream\t%sName%s\n", blueColor, yellowColor, greenColor, noColor)
+	fmt.Fprintf(os.Stderr, "%s================================================================================%s\n", grayColor, noColor)
+
+	for _, proxyName := range proxyNames {
+		proxy := proxies[proxyName]
+		fmt.Printf("%s%s\t%s%s\t%s%s%s\n", blueColor, proxy.Listen, yellowColor, proxy.Upstream, enabledColor(proxy.Enabled), proxy.Name, noColor)
+	}
+}
+func inspect(c *cli.Context, t *toxiproxy.Client) {
+	proxyName := c.Args().First()
+
+	proxy, err := t.Proxy(proxyName)
+	if err != nil {
+		fmt.Printf("Failed to retrieve proxy %s: %s\n", proxyName, err.Error())
+		os.Exit(1)
+	}
+
+	// TODO It would be cool to include the enabled toxics in the pipe
+	// rendering --L-SC-> for latency + slow close enabled
+	fmt.Printf("%s%s%s\n", enabledColor(proxy.Enabled), proxy.Name, noColor)
+	fmt.Printf("%s%s %s---> %s%s%s\n\n", blueColor, proxy.Listen, grayColor, yellowColor, proxy.Upstream, noColor)
+
+	listToxics(proxy.ToxicsUpstream, "upstream")
+	fmt.Println()
+	listToxics(proxy.ToxicsDownstream, "downstream")
+}
+
+func toggle(c *cli.Context, t *toxiproxy.Client) {
+	proxyName := c.Args().First()
+
+	proxy, err := t.Proxy(proxyName)
+	if err != nil {
+		fmt.Printf("Failed to retrieve proxy %s: %s\n", proxyName, err.Error())
+		os.Exit(1)
+	}
+
+	proxy.Enabled = !proxy.Enabled
+
+	err = proxy.Save()
+	if err != nil {
+		fmt.Printf("Failed to toggle proxy %s: %s\n", proxyName, err.Error())
+		os.Exit(1)
+	}
+
+	fmt.Printf("Proxy %s%s%s is now %s%s%s\n", enabledColor(proxy.Enabled), proxyName, noColor, enabledColor(proxy.Enabled), enabledText(proxy.Enabled), noColor)
+}
+
+func create(c *cli.Context, t *toxiproxy.Client) {
+	proxyName := c.Args().Get(0)
+	listen := c.Args().Get(1)
+	upstream := c.Args().Get(2)
+	p := &toxiproxy.Proxy{
+		Name:     proxyName,
+		Listen:   listen,
+		Upstream: upstream,
+		Enabled:  true,
+	}
+	p = t.NewProxy(p)
+	err := p.Create()
+	if err != nil {
+		log.Fatalf("Failed to create proxy: %s\n",
+			err.Error())
+	}
+	fmt.Printf("Created new proxy %s\n", proxyName)
+}
+
+func delete(c *cli.Context, t *toxiproxy.Client) {
+	proxyName := c.Args().First()
+	p, err := t.Proxy(proxyName)
+	if err != nil {
+		fmt.Printf("Failed to retrieve proxy %s: %s\n", proxyName, err.Error())
+		os.Exit(1)
+	}
+
+	err = p.Delete()
+	if err != nil {
+		log.Fatalf("Failed to delete proxy: %s\n",
+			err.Error())
+	}
+	fmt.Printf("Deleted proxy %s\n", proxyName)
+}
+
+func addToxic(c *cli.Context, t *toxiproxy.Client) {
+	proxyName := c.Args().Get(0)
+	toxicName := c.Args().Get(1)
+	direction := c.Args().Get(2)
+	toxicConfig := c.Args().Get(3)
```

Comment:
I think we should use named arguments instead. How does this look in the help pages?